packages/components/src/components/linear-progress/linear-progress-container.jsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { LinearProgress } from 'Components/linear-progress/linear-progress.jsx';
+
+class LinearProgressContainer extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            timeout: props.timeout / 1000,
+            total_ticks: Math.round(props.timeout / 1000),
+            current_tick: Math.round(props.timeout / 1000),
+        };
+    }
+
+    get progress() {
+        return 100 - Math.round((this.state.current_tick / this.state.total_ticks) * 100);
+    }
+
+    get remaining() {
+        return this.state.timeout >= 0 ? this.state.timeout : 0;
+    }
+
+    makeProgress = () => {
+        this.setState({
+            current_tick: this.state.current_tick - 1,
+            timeout: this.state.timeout - 1,
+        });
+    };
+
+    run = () => {
+        this.props.action();
+    };
+
+    componentDidMount() {
+        this.interval = setInterval(this.makeProgress, 1000);
+    }
+
+    componentDidUpdate() {
+        if (this.progress > 100) {
+            this.run();
+        }
+    }
+
+    componentWillUnmount() {
+        clearInterval(this.interval);
+    }
+
+    render() {
+        const { timeout, className } = this.props;
+        if (!timeout) return null;
+        // return null;
+        return (
+            <div className='notification__timeout-container'>
+                <div className='notification__timeout-countdown'>{this.props.render(this.remaining)}</div>
```

Comment:
These classes are relying on styles that are defined in another component in another package. Is there any better way to handle this? Perhaps we can add the styles it needs to its own stylesheet?

Reply:
You are right, I'll refactor it right away.